torch/_inductor/pattern_matcher.py

```diff
@@ -907,6 +907,12 @@ def normalize_args(**kwargs):
         assert not kwargs, f"leftover kwargs: {kwargs!r}"
         return args
 
+    if trace_fn is training_graph:
+        # If inference mode is enabled during compilation, assume that we don't
+        # want to match on any training graph patterns
+        if torch.is_inference_mode_enabled():
+            return False
```

Comment:
cc @Chillee. More details in the PR description, but let me know what you think.

Reply:
Hmm... this is a bit weird to me, but I guess it's fine.

In particular, there's no reason an "inference" graph can't also have joint graph patterns apply to it. For example, if we capture forwards and backwards in a single graph, we might go down the inference graph path anyways?

I guess this wouldn't work today anyways, as these passes are hardcoded to trigger in the "joint graph" part. But something to keep in mind in the future.